evap/staff/staff_mode.py

```diff
@@ -52,6 +52,9 @@ def is_in_staff_mode(request):
 
 
 def update_staff_mode(request):
+    if not request.user.has_staff_permission: 
+        exit_staff_mode(request)
+        return
     assert request.user.has_staff_permission
```

Comment:
I think this assertion is no longer needed

Reply:
I'd move it up to line 33, where I think it would still add value.

The call graphs we can get now are a bit wonky. We can have
`enter_staff_mode` -> `update_staff_mode` -> finds out that `not request.user.has_staff_permission` -> `exit_staff_mode` and returns without exception, which is probably never what the initial caller intended?

It seems to me that `update_staff_mode` was about "update the timestamp in the session to now()" in the past, now it has changed semantics. Could use a little clean up in general, I think.